src/btree/row_modify.c

```diff
@@ -255,14 +255,21 @@ __wt_row_insert_alloc(WT_SESSION_IMPL *session,
  *	Allocate a WT_UPDATE structure and associated value and fill it in.
  */
 int
-__wt_update_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value,
-    WT_UPDATE **updp, size_t *sizep, u_int modify_type)
+__wt_update_alloc(WT_SESSION_IMPL *session,
+    const WT_ITEM *value, WT_UPDATE **updp, size_t *sizep, int modify_type)
 {
 	WT_UPDATE *upd;
 
 	*updp = NULL;
 
 	/*
+	 * The code paths leading here are convoluted: assert we never attempt
+	 * to allocate an update structure if only intending to insert one we
+	 * already have.
+	 */
+	WT_ASSERT(session, modify_type != -1);
```

Comment:
I don't feel super strongly, but seeing a whole bunch of `-1` sprinkled around felt weird to me.  Looking through our code for that string, the vast, vast majority of them are in the capacity of a `cmp` of some sort which makes sense.  Most non-`cmp` uses seem to be in the `os_*` directories.